templates/news-detail/news-detail.js

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable function-paren-newline, object-curly-newline */
+import { div, h2, a, img } from '../../scripts/dom-helpers.js';
+import { getMetadata } from '../../scripts/aem.js';
+import { breadcrumbs } from '../../scripts/breadcrumbs.js';
+
+export default function decorate(doc) {
+  const $main = doc.querySelector('main');
+  const $picture = $main.querySelector('picture');
+  const $hero = div({ class: 'hero' },
+    $picture,
+  );
+  // const $content = Array.from(main.querySelectorAll('.default-content-wrapper *'));
+  const $content = $main.querySelector('.default-content-wrapper');
+  const teaserTitle = getMetadata('teaser-title');
+  const teaserDescription = getMetadata('teaser-description');
+  const type = getMetadata('type');
+  const publishedDate = getMetadata('published-date');
+  const categories = getMetadata('categories');
+  const $breadcrumbs = breadcrumbs();
+
+  const $header = div({ class: 'header' },
+    $breadcrumbs,
+    div({ class: 'type' }, type),
+    h2(teaserTitle),
+    div({ class: 'category-tags' },
+      publishedDate,
+      ' | ',
+      categories,
+    ),
+    div({ class: 'description' }, teaserDescription),
+    div({ class: 'social-share' },
+      a({
+        class: 'icon icon-linkedin',
```

Comment:
Should we take icon and href from authoring in metadata.? Currently this is hardcoded for all.

Reply:
This seemed to be consistent across all news articles.  I couldn't find an example where it was anything but linkedin.  We can ask the customer if they would like this author able.

Reply:
Found multiple social icons in another locale: https://www.ingredion.com/apac/sc-cn/news-events/news/ingredion-resetting-market-sustainably-sourced-ingredients.html

Also, I believe we can hardcode the href basis the social sites and append page url to it. I checked the functionality, it's basically for creating a post on these sites for this news article.

Linkedin: "https://www.linkedin.com/cws/share?url=" + page url
Facebook: "http://www.facebook.com/sharer.php?u=" + page url
etc

Reply:
Ok, good catch. Let me rework this

Reply:
Updated - https://news-detail--ingredion--aemsites.aem.live/na/en-us/news-events/news/kerr-by-ingredion-announces-upcycled-certification-html
also converted twitter to x